api/src/services/applications/update_application_form.py

```diff
@@ -72,23 +82,42 @@ def update_application_form(
 
     if application_form:
         # Update existing application form
-        application_form.application_response = application_response
+        if application_response is not None:
+            application_form.application_response = application_response
+        if is_included_in_submission is not None:
+            application_form.is_included_in_submission = is_included_in_submission
     else:
-        # Create new application form
+        # Create new application form (requires application_response)
+        if application_response is None:
+            raise_flask_error(
+                404,
+                f"Application form not found for application {application_id} and form {form_id}",
+            )
+
         application_form = ApplicationForm(
             application=application,
             competition_form=competition_form,
             application_response=application_response,
+            is_included_in_submission=is_included_in_submission,
         )
         db_session.add(application_form)
 
-    # Get a list of validation warnings (also sets form status)
-    warnings: list[ValidationErrorDetail] = validate_application_form(
-        application_form, ApplicationAction.MODIFY
-    )
+    # Only validate if we're updating form content
+    if application_response is not None:
+        warnings: list[ValidationErrorDetail] = validate_application_form(
+            application_form, ApplicationAction.MODIFY
+        )
+    else:
+        warnings = []
+
```

Comment:
We should always do the validation, even if they're not changing their answers.

Because of how the validation will work, if a form is marked as "included" that changes the validation as we suddenly want to enforce it, so even if their answers didn't change, the way we validate does.